homeassistant/components/nextbus/config_flow.py

```diff
@@ -0,0 +1,211 @@
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_AGENCY, CONF_ROUTE, CONF_STOP, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _dict_to_select_selector(options: dict[str, str]) -> SelectSelector:
+    return SelectSelector(
+        SelectSelectorConfig(
+            options=[
+                SelectOptionDict(value=key, label=value)
+                for key, value in options.items()
+            ],
+            mode=SelectSelectorMode.DROPDOWN,
+        )
+    )
+
+
+def _get_agency_tags(client: NextBusClient) -> dict[str, str]:
+    return {a["tag"]: a["title"] for a in client.get_agency_list()["agency"]}
+
+
+def _get_route_tags(client: NextBusClient, agency_tag: str) -> dict[str, str]:
+    return {a["tag"]: a["title"] for a in client.get_route_list(agency_tag)["route"]}
+
+
+def _get_stop_tags(
+    client: NextBusClient, agency_tag: str, route_tag: str
+) -> dict[str, str]:
+    return {
+        a["tag"]: a["title"]
+        for a in client.get_route_config(route_tag, agency_tag)["route"]["stop"]
+    }
+
+
+def _validate_import(
+    client: NextBusClient, agency: str, route: str, stop: str
+) -> str | None:
+    agency_tags = _get_agency_tags(client)
+    if agency not in agency_tags:
+        return "invalid_agency"
+
+    route_tags = _get_route_tags(client, agency)
+    if route not in route_tags:
+        return "invalid_route"
+
+    stop_tags = _get_stop_tags(client, agency, route)
+    if stop not in stop_tags:
+        return "invalid_stop"
+
+    return None
+
+
+class NextBusFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle Nextbus configuration."""
+
+    VERSION = 1
+
+    _agency_tags: dict[str, str]
+    _route_tags: dict[str, str]
+    _stop_tags: dict[str, str]
+
+    def __init__(self):
+        """Initialize NextBus config flow."""
+        self.data: dict[str, str] = {}
+        self._client = NextBusClient(output_format="json")
+        _LOGGER.info("Init new config flow")
+
+    async def async_step_import(self, config_input: dict[str, str]) -> FlowResult:
+        """Handle import of config."""
+        agency = config_input[CONF_AGENCY]
+        route = config_input[CONF_ROUTE]
+        stop = config_input[CONF_STOP]
+
+        if validation_error := await self.hass.async_add_executor_job(
+            _validate_import,
+            self._client,
+            agency,
+            route,
+            stop,
+        ):
+            return self.async_abort(reason=validation_error)
+
+        data = {
+            CONF_AGENCY: agency,
+            CONF_ROUTE: route,
+            CONF_STOP: stop,
+        }
+
+        # Abort if duplicate entries exist
+        self._async_abort_entries_match(data)
```

Comment:
```suggestion
        await self.async_set_unique_id("_".join(value for _, value in data.items()))
        self._abort_if_unique_id_configured()
```

I discussed it with @frenck, and you can use `agency + route + stop` as a unique ID for this integration.

Reply:
Thanks! That's great news. 👍 I'll refactor that back in.